raster/src/main/scala/geotrellis/raster/merge/SinglebandTileMergeMethods.scala

```diff
@@ -154,4 +154,69 @@ trait SinglebandTileMergeMethods extends TileMergeMethods[Tile] {
       case _ =>
         self
     }
+
+  def union(extent: Extent, otherExtent: Extent, other: Tile, method: ResampleMethod, unionF: (Option[Double], Option[Double]) => Double): Tile = {
+    val unionInt = (l: Option[Double], r: Option[Double]) => unionF(l, r).toInt
+
+    val combinedExtent = otherExtent combine extent
+    val re = RasterExtent(extent, self.cols, self.rows)
+    val gridBounds = re.gridBoundsFor(combinedExtent, false)
+    val targetCS = CellSize(combinedExtent, gridBounds.width, gridBounds.height)
+    val targetRE = RasterExtent(combinedExtent, targetCS)
+    val mutableTile = ArrayTile.empty(self.cellType, targetRE.cols, targetRE.rows)
+
+    self.cellType match {
+      case BitCellType | ByteCellType | UByteCellType | ShortCellType | UShortCellType | IntCellType  =>
+        val interpolateLeft: (Double, Double) => Int = Resample(method, self, extent, targetCS).resample _
+        val interpolateRight: (Double, Double) => Int = Resample(method, other, otherExtent, targetCS).resample _
+        // Assume 0 as the transparent value
+        cfor(0)(_ < targetRE.rows, _ + 1) { row =>
+          cfor(0)(_ < targetRE.cols, _ + 1) { col =>
+            val (x,y) = targetRE.gridToMap(col, row)
+            val (l,r) = (interpolateLeft(x, y), interpolateRight(x, y))
+            mutableTile.set(col, row, unionInt(Some(l.toDouble), Some(r.toDouble)))
+          }
+        }
+      case FloatCellType | DoubleCellType =>
+        val interpolateLeft: (Double, Double) => Double = Resample(method, self, extent, targetCS).resampleDouble _
+        val interpolateRight: (Double, Double) => Double = Resample(method, other, otherExtent, targetCS).resampleDouble _
+
+        // Assume 0.0 as the transparent value
+        cfor(0)(_ < targetRE.rows, _ + 1) { row =>
+          cfor(0)(_ < targetRE.cols, _ + 1) { col =>
+            val (x,y) = targetRE.gridToMap(col, row)
+            val (l,r) = (interpolateLeft(x, y), interpolateRight(x, y))
+            mutableTile.setDouble(col, row, unionF(Some(l), Some(r)))
```

Comment:
`isData` handling as well? (in case it's not implied that it's incapuslated into the union function definition)